internal/integration/rhp/v4/rhp4_test.go

```diff
@@ -0,0 +1,1383 @@
+	cm := hn.Chain
+	w := hn.Wallet
+
+	results := make(chan error, 1)
+	if _, err := hn.Volumes.AddVolume(context.Background(), filepath.Join(t.TempDir(), "test.dat"), 64, results); err != nil {
+		t.Fatal(err)
+	} else if err := <-results; err != nil {
+		t.Fatal(err)
+	}
+
+	testutil.MineAndSync(t, hn, w.Address(), int(n.MaturityDelay+20))
+
+	transport := testRenterHostPair(t, hostKey, hn, zap.NewNop())
+
+	settings, err := rhp4.RPCSettings(context.Background(), transport)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fundAndSign := &fundAndSign{w, renterKey}
+	renterAllowance, hostCollateral := types.Siacoins(100), types.Siacoins(200)
+	formResult, err := rhp4.RPCFormContract(context.Background(), transport, cm, fundAndSign, cm.TipState(), settings.Prices, hostKey.PublicKey(), settings.WalletAddress, proto4.RPCFormContractParams{
+		RenterPublicKey: renterKey.PublicKey(),
+		RenterAddress:   w.Address(),
+		Allowance:       renterAllowance,
+		Collateral:      hostCollateral,
+		ProofHeight:     cm.Tip().Height + 50,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	revision := formResult.Contract
+
+	cs := cm.TipState()
+	account := proto4.Account(renterKey.PublicKey())
+
+	accountFundAmount := types.Siacoins(25)
+	fundResult, err := rhp4.RPCFundAccounts(context.Background(), transport, cs, renterKey, revision, []proto4.AccountDeposit{
+		{Account: account, Amount: accountFundAmount},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	revision.Revision = fundResult.Revision
+
+	token := proto4.AccountToken{
+		Account:    account,
+		ValidUntil: time.Now().Add(time.Hour),
+	}
+	tokenSigHash := token.SigHash()
+	token.Signature = renterKey.SignHash(tokenSigHash)
+
+	roots := make([]types.Hash256, 0, 50)
+
+	checkRoots := func(t *testing.T, expected []types.Hash256) {
+		t.Helper()
+
+		rootsResult, err := rhp4.RPCSectorRoots(context.Background(), transport, cs, settings.Prices, renterKey, revision, 0, uint64(len(expected)))
+		if err != nil {
+			t.Fatal(err)
+		} else if len(roots) != len(expected) {
+			t.Fatalf("expected %v roots, got %v", len(expected), len(roots))
+		}
+		for i := range rootsResult.Roots {
+			if roots[i] != expected[i] {
+				t.Fatalf("expected %v, got %v", expected[i], roots[i])
+			}
+		}
+		revision.Revision = rootsResult.Revision
+	}
+
+	for i := 0; i < cap(roots); i++ {
+		// store random sectors on the host
+		data := frand.Bytes(1024)
+
+		// store the sector
+		writeResult, err := rhp4.RPCWriteSector(context.Background(), transport, settings.Prices, token, bytes.NewReader(data), uint64(len(data)))
+		if err != nil {
+			t.Fatal(err)
+		}
+		roots = append(roots, writeResult.Root)
+
+		appendResult, err := rhp4.RPCAppendSectors(context.Background(), transport, cs, settings.Prices, renterKey, revision, []types.Hash256{writeResult.Root})
+		if err != nil {
+			t.Fatal(err)
+		}
+		revision.Revision = appendResult.Revision
+		checkRoots(t, roots)
+	}
+}
+
+func BenchmarkWrite(b *testing.B) {
```

Comment:
`nit` makes sense to add whatever result you get here on your machine

Ran them on mine in case you want to just grab that one:
```
goos: darwin
goarch: arm64
pkg: go.sia.tech/hostd/internal/integration/rhp/v4
cpu: Apple M1 Max
=== RUN   BenchmarkWrite
BenchmarkWrite
BenchmarkWrite-10             24          50311752 ns/op          83.37 MB/s     5015286 B/op      12820 allocs/op
PASS
ok      go.sia.tech/hostd/internal/integration/rhp/v4   5.204s
```